posture/serialnumber_stub.go

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !darwin
```

Comment:
I think it might be simpler to call this `serialnumber.go` and remove the build tag.

Reply:
The way we usually approach this I think is by defining a function variable in the non-platform-specific file, and setting it per platform ([example](https://github.com/search?q=repo%3Atailscale%2Ftailscale%20newOSImpl&type=code)). This makes a pretty logical structure: you can often see what platforms are supported just by looking at the list of files, and can return an error on unsupported platforms without maintaining a negative build tag.

Reply:
We [do both](https://github.com/search?q=repo%3Atailscale%2Ftailscale+_stub&type=code), I do not have a strong opinion, I can change the implementation to use the register approach.

Reply:
Oh nice, we have two different standards :) In that case I don't really care either.